src/m365/pa/commands/app/app-remove.ts

```diff
@@ -0,0 +1,117 @@
+import * as chalk from 'chalk';
+import { Cli, Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import AzmgmtCommand from '../../../base/AzmgmtCommand';
+import commands from '../../commands';
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  environment: string;
+  name: string;
+  asAdmin?: boolean;
+  confirm?: boolean;
+}
+
+class PaAppRemoveCommand extends AzmgmtCommand {
+  public get name(): string {
+    return commands.APP_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the specified Microsoft Power App';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.asAdmin = typeof args.options.asAdmin !== 'undefined';
+    telemetryProps.confirm = typeof args.options.confirm !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Removing Microsoft Power App ${args.options.name}...`);
+    }
+
+    const removePaApp: () => void = (): void => {
+      const requestOptions: any = {
+        url: `${this.resource}providers/Microsoft.PowerApps${args.options.asAdmin ? `/scopes/admin/environments/${encodeURIComponent(args.options.environment)}` : ''}/apps/${encodeURIComponent(args.options.name)}?api-version=2016-11-01`,
```

Comment:
Have you tried using the `https://management.azure.com/providers/Microsoft.PowerApps/apps/<name>?api-version=2017-08-01` API? I saw that you can use it to delete the app you own. If it's possible to use it delete apps from others as well, it would be better to use this API, because it doesn't require retrieving the environment ID. It would be also more in line with the existing `pa app list` command which doesn't require the environment ID either.

Reply:
Hey @waldekmastykarz I did some digging and found indeed that you can remove Power Apps from others using this 2017-08-01 API version. That kind of makes the `--asAdmin` obsolete don't you agree? How did you found out about this? I can't find anything documented. Just the version list of available API versions. I've tried the proces using the latest version 2021-02-01, which also works. Would it be a good idea to just use the latest version?

Reply:
@waldekmastykarz did you also check if you could delete apps from another environment with that API? As you can have multiple environments (as a user you can be part of multiple environment) (before we remove the `environment` option).

Reply:
Nope, I haven't. That's what we'd need to verify. If that's not possible you need to specify the environment, then we should consider adding an `pa environment list` alias to `flow environment list` so that folks can get environments.

Reply:
We have the get for both environment options already so we are good there!

Reply:
Ah check, I thought we missed it. My bad

Reply:
Hi @appieschot, @waldekmastykarz, as a user with Maker role I was able to delete a PowerApp from different environments without specifying the environment ID. Same as Administrator. It seems to me that Power App Name therefor is a unique identifier across environments. Is that safe to assume? In that case I can push my changes that omit `--asAdmin` and `--environment`.

Reply:
Sounds like a plan! Thanks for testing